Fix Bug 1140186 - Update meta tags on Firefox pages

kyoshino opened the pull request.

flodolo commented on bedrock/firefox/templates/firefox/developer.html:
```diff
@@ -6,7 +6,16 @@
+{% block page_image %}{{ static('img/firefox/developer/page-image.png') }}{% endblock %}
+
+{% block page_desc %}
+  {% if l10n_has_tag('meta_update_201403') %}
```
You really need to convince yourself that we're in 2015 ;-)

kyoshino replied:
Ahhhhh. I just don't want to get old  :stuck_out_tongue_closed_eyes:

alexgibson commented on bedrock/firefox/templates/firefox/privacy_tour/index.html:
```diff
@@ -11,6 +11,20 @@
 
 {% block extra_meta %}<meta name="robots" content="noindex">{% endblock %}
 
+{% block page_title %}
```
I'm a bit confused as to why we're updating strings on this page? The privacy tour was shown as a whatsnew page for users updating to 33.1, which shipped quite a while ago. I doubt there's much traffic to this page (if any at all). It's also a page only viewable by FF users and is not indexed by search engines, so I'm not sure what benefit there is here.

alexgibson replied:
Come to think of it, would the `firefox/independent` page be a better fit for this, since that was the publically viewable campaign page for this release?

kyoshino replied:
Yes, the whatnew pages are not intended to be shared. I'll ask in the bug.

alexgibson replied:
I would recommend removing this from the PR, as it's just creating superfluous changes.

alexgibson commented on bedrock/firefox/templates/firefox/dev-firstrun.html:
```diff
@@ -15,6 +15,8 @@
 {% block page_title_prefix %}{% endblock %}
 {% block page_title %}{{_('Welcome to Firefox Developer Edition')}}{% endblock %}
 {% block page_title_suffix %}{% endblock %}
+{% block page_image %}{{ static('img/firefox/developer/page-image.png') }}{% endblock %}
```
There's no harm in adding this image here, but `/firstrun` pages should not form part of a social campaign, since they are treated as in-product pages only accessible to Firefox. Probably good to make sure folks understand this when asking for changes.

alexgibson replied:
Since this is a `/firstrun` page we should probably override `{% block page_og_url %}` too.

alexgibson replied:
Oh, nevermind - that's already done here below :+1:

alexgibson commented on bedrock/firefox/templates/firefox/all.html:
```diff
@@ -13,7 +13,15 @@
     {{ _('Firefox ESR is intended for groups who deploy and maintain the desktop environment in large organizations. <a href="%s">Learn more.</a>')|format(url('firefox.organizations.organizations')) }}
   {% else %}
-    {{ _('Firefox is available in more than 80 languages thanks to the contributions of Mozilla community members from around the world.') }}
+    {{ _('Firefox is made in large part by volunteers around the world. That’s why we offer it in 80 languages (and why there’s heart and soul in every piece of code).') }}
```
This is a new string not in an l10n tag, but I'm assuming this is ok and you are handling the `ja` locale?

The current one is still untranslated on prod since the last change: https://www.mozilla.org/ja/firefox/all/

flodolo replied:
Japanese is the only localization for this page, I assume @kyoshino is already on this one?

kyoshino replied:
Yes, I can update the Japanese strings myself.

alexgibson commented:
https://bugzilla.mozilla.org/show_bug.cgi?id=1140186#c8

"Please don't update the title on /firefox/new/ as that one has already been optimizations for SEO. All other pages, I am fine with. /firefox/new/ is critical to optimize to seo and not brand terms."

alexgibson commented on bedrock/firefox/templates/firefox/new.html:
```diff
@@ -6,7 +6,22 @@
 {% block page_title %}{{_('Free Web Browser')}}{% endblock %}
-{% block page_desc %}{{_('Download Mozilla Firefox, a free Web browser. Firefox is created by a global non-profit dedicated to putting individuals in control online. Get Firefox today!')}}{% endblock %}
+
+{#- This will appear as <meta property="og:title"> which can be used for social share -#}
```
https://bugzilla.mozilla.org/show_bug.cgi?id=1140186#c8

"Please don't update the title on /firefox/new/ as that one has already been optimizations for SEO. All other pages, I am fine with. /firefox/new/ is critical to optimize to seo and not brand terms."

kyoshino replied:
The `page_og_title` block is only for `<meta property="og:title">`. The `<title>` will remain as is.

alexgibson replied:
The page description is also changing here though? Will this likely also impact SEO on this page?

kyoshino replied:
Ahh. That's why I myself have introduced the `page_og_desc` block as well...

kyoshino commented:
Discarded the changes to the firstrun and whatsnew pages. Nothing harmful but useless.

flodolo commented on bedrock/firefox/templates/firefox/whatsnew-base-resp.html:
```diff
@@ -0,0 +1,18 @@
+{#- This will appear as <meta property="og:description"> which can be used for social share -#}
+{% block page_og_desc %}
+  {% if l10n_has_tag('meta_update_201503') %}
+    {{_('Firefox is non-profit, non-corporate, non-compromised. Choosing Firefox isn’t just choosing a browser. It’s a vote for personal freedom.')}}
```
This doesn't make sense. Why are we adding this string to a base template? Base template should not rely on l10n tags either.

Same for the next file. If it's something we need, we'll have to add it to a shared file like main.lang

alexgibson replied:
I'm confused why we're now adding these to all whatsnew pages? As previously mentioned, these should be treated as internal FF only pages and are not really meant for sharing. Much of the code for many of these pages is only targeted for specific versions of Firefox.

kyoshino replied:
Looks like I have force pushed my WIP, but... because there are many whatsnew and firstrun templates and I didn't want to add the string to each template. That string can go to `main.lang` I guess.

alexgibson replied:
Kohei, are these changes really necessary in these pages?

kyoshino replied:
@alexgibson according to discussion in the bug, those in-product URLs may also be shared by visitors even though those are dead-end pages, so I have proposed a solution of overriding `og:url` to make the pages shareable.

flodolo replied:
To be completely honest, I don't like the approach. We can add the string to main.lang, and then we're stuck next time we decide to update this string because we want to focus communication on something else.

I'd prefer what's new page/first run to be threated individually, and add string as necessary. I'd also like to see numbers on how many people actually share the whatsnew page (do we have them?), because it really seems an edge case.

alexgibson replied:
Regardless as to whether some people may share them, I don't think it is something we should necessarily encourage :/

alexgibson replied:
I've commented in the bug for discussion, thanks @kyoshino

kyoshino replied:
Um, yeah. I personally think we should rather add a Share button to the whatsnew/firstrun pages to encourage people to tweet "Hey I just installed Firefox!" or something else, but these `og` tags might not be necessary anyway, because Share buttons can have an arbitrary string and URL.

alexgibson replied:
Ah, @kyoshino I see you have changed the sharing URL for these pages? That makes me not so worried :+1: 

Still, I think Flod's comments about this being an edge case still warrant some investigation.

alexgibson replied:
Well, seeing as there's probably very little traffic to these pages anyway, changing them now still seems maybe unnecessary.

alexgibson replied:
@kyoshino why not just add these meta tags to the most recent /whatsnew and /firstrun? Seems that would simplify both the string situation and not make the templates less confusing?

kyoshino replied:
Actually I cannot recognize which are the latest whatsnew/firstrun pages :confused: And if new variations of those pages is added, we could miss the string, so I thought the intermediate `whatsnew-base` template was a good idea.

alexgibson replied:
Using a base template for these strings might be a fine enough idea, but it just seems like changing all these old templates is a wasted effort?

Whatsnew: https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/views.py#L472

Firstrun: https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/views.py#L418

alexgibson commented on bedrock/firefox/templates/firefox/nightly_firstrun.html:
```diff
@@ -4,9 +4,7 @@
-{% extends "/firefox/base-resp.html" %}
-
-{% block extra_meta %}<meta name="robots" content="noindex">{% endblock %}
+{% extends "/firefox/whatsnew-base-resp.html" %}
```
Also seems confusing that firstrun templates are now extending `whatsnew-base-resp.html`. Should probably find a better name?

alexgibson commented on bedrock/firefox/templates/firefox/nightly_firstrun.html:
```diff
@@ -17,6 +15,8 @@
 {% block page_title_prefix %}{% endblock %}
 {% block page_title %}{{ _('Firefox Nightly First Run Page') }}{% endblock %}
 {% block page_title_suffix %}{% endblock %}
+{% block page_og_url %}http://nightly.mozilla.org/{% endblock %}
```
`https://`

kyoshino commented:
Sorry for the delay here. I have removed the intermediate whatsnew templates and added the `page_og` blocks only to the recent whatsnew/firstrun pages. Does it work? @flodolo

alexgibson commented on bedrock/firefox/templates/firefox/privacy_tour/index.html:
```diff
@@ -11,6 +11,23 @@
 
 {% block extra_meta %}<meta name="robots" content="noindex">{% endblock %}
 
+{#- This will appear as <meta property="og:title"> which can be used for social share -#}
```
This is an old page, so this one can probably just be removed.

kyoshino replied:
Will remove it.

alexgibson commented on bedrock/firefox/templates/firefox/search_tour/index.html:
```diff
@@ -6,6 +6,23 @@
 
 {% block extra_meta %}<meta name="robots" content="noindex">{% endblock %}
 
+{#- This will appear as <meta property="og:title"> which can be used for social share -#}
```
Same here

alexgibson commented on bedrock/mozorg/helpers/misc.py:
```diff
@@ -462,8 +462,10 @@ def absolute_url(url):
     if url.startswith('//'):
         prefix = 'https:'
-    else:
+    elif url.startswith('/'):
```
This is only a tiny change and has a test below, so looks OK to me. But I'd appreciate it if either @pmclanahan or @jgmize could take a quick look and give their blessing.

pmclanahan replied:
I actually don't see a test for the new "else" branch of this code. Should have a unit test in there to ensure what you expect happens when the function is passed something that doesn't start with a slash.

kyoshino replied:
Umm, the added `test_full` didn't make sense. I have added `test_urls` instead to check URLs as intended.

alexgibson replied:
@pmclanahan appreciative if you could take a quick look at the updated tests here :+1:

pmclanahan replied:
APPROVED :+1:

flodolo commented on bedrock/firefox/templates/firefox/australis/fx36/tour-base.html:
```diff
@@ -8,6 +8,23 @@
 
+{#- This will appear as <meta property="og:title"> which can be used for social share -#}
+{% block page_og_title %}
+  {% if l10n_has_tag('meta_update_201503') %}
```
Since we're using these 2 strings in 3 files, it makes more sense to have them in a shared file (and we can't have tags on those).

So, let's remove the l10n_has_tag conditional on these pages
```
bedrock/firefox/templates/firefox/australis/fx36/tour-base.html
bedrock/firefox/templates/firefox/australis/tour-base.html
bedrock/firefox/templates/firefox/new.html
```
I'll move the strings currently in firefox/new.lang into main.lang

kyoshino replied:
Thanks @flodolo, `main.lang` might be the best place for those strings.

alexgibson replied:
@kyoshino - I think we're good to merge this one once we remove the l10n_has_tag conditional in the pages @flodolo suggests. Thanks for all your work on this one!

kyoshino replied:
Removed the `l10n_has_tag` conditional.
